modules/apps/dataset/dataset-api/src/main/java/com/liferay/dataset/filter/DatasetFilter.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dataset.filter;
+
+/**
+ * @author Marco Leo
+ */
+public interface DatasetFilter {
```

Comment:
We have both a `DatasetDataFilter` and a `DatasetFilter`... maybe this one is one we could differentiate somehow? Is this a UI filter? maybe a `dataset.ui` package would be a better fit?

Reply:
This is the thing I was talking about when I said "mixing data and GUI" in Slack. I put all the raw data stuff in the `data` subpackage, leaving the rest for GUI stuff. However that leads to:

1) duplicity (in this case)
2) some blur, because we have, for example, `data` and `filter` or `action` at the same level.

Maybe we can do one more package level to differentiate between the GUI component and its backend :thinking: . Maybe the problem is that we are using the `-api` bundle for two different things... I don't think we have any other place in portal where an API serves a GUI component + a backend service :thinking:

Reply:
I'm going to create the `ui` package and move things there to see how it looks. Will push the commit.